lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1332,9 +1332,16 @@ static void addLinkOptionsPostorder(CodeGenModule &CGM, Module *Mod,
     // Link against a framework.  Frameworks are currently Darwin only, so we
     // don't to ask TargetCodeGenInfo for the spelling of the linker option.
     if (Mod->LinkLibraries[I-1].IsFramework) {
+      StringRef LibName = Mod->LinkLibraries[I - 1].Library;
+
+      // For modules that use export_as for linking, use that module
+      // name instead.
+      if (Mod->Name == LibName && Mod->UseExportAsModuleLinkName)
```

Comment:
Why limit this to `Mod->Name == LibName`? Is this to handle private modules?

Reply:
'use' declarations can add more libraries, it seems to me those should still show up.

Reply:
I'm not sure I agree. If we ever extended this to non-frameworks, you'd definitely *not* want them to show up (because that would defeat the purpose).

Reply:
What happens if SomeKitCore is declaring use of module X that SomeKit isn't declaring in its module map. We don't X to show up as -framework X even if we are only importing SomeKit?

Reply:
That sounds correct to me. `export_as` means that anything in SomeKitCore is going to be visible in SomeKit, without exception, which means that SomeKit must itself declare all the same `use` flags to link successfully.

(If we allowed `export_as` in non-top-level modules this would get trickier.)

@DougGregor, what do you think?

Reply:
I agree with @jrose-apple that we should not restrict to this specific module name. The re-exporting module `SomeKit` should be responsible for re-declaring any `use` flags from the modules it re-exports, because this is the module that's ultimately responsible for the full interface and it has to stand alone: the module with `export_as` (`SomeKitCore` in this example) might not be around in some configurations, so we can't rely on gathering information from its module map.